nixos/modules/system/boot/systemd/tmpfiles.nix

```diff
@@ -100,6 +153,18 @@ in
           ${concatStringsSep "\n" cfg.rules}
         '';
       })
+      (pkgs.runCommand "systemd-extra-tmpfiles" {} ''
+        mkdir -p $out/lib/tmpfiles.d
+        cd $out/lib/tmpfiles.d
+
+        ${concatMapStrings (namespace:
+          concatMapStrings (path:
+            concatMapStrings (entry: ''
+              echo "${entry.type} ${path} ${entry.mode} ${entry.user} ${entry.group} ${entry.age} ${entry.argument}" >> nixos-${namespace}.conf
```

Comment:
if we're leaving quoting of arguments to the user we should explicitly mention this... or just take care of it for them

Reply:
how to properly quote them? can we wrap them in " quotes?

Reply:
seems that it works for my simple test case with single quotes

Reply:
https://www.freedesktop.org/software/systemd/man/latest/tmpfiles.d.html

> Fields may contain C-style escapes. With the exception of the seventh field (the "argument") all fields may be enclosed in quotes. Note that any whitespace found in the line after the beginning of the argument field will be considered part of the argument field. To begin the argument field with a whitespace character, use C-style escapes (e.g. "\x20").

ok so we should quote all fields except the last one, `argument`